pkg/cmd/adm/register_member.go

```diff
@@ -29,93 +39,131 @@ const (
+			return err
+		}
+
+		*v.uint = uint(vi)
+	}
+	return nil
+}
+
+func (v optionalUint) String() string {
+	if v.uint != nil {
+		return fmt.Sprintf("%d", *v.uint)
+	} else {
+		return "<unspecified>"
+	}
+}
+
+func (v optionalUint) Type() string {
+	return "uint"
+}
+
 func NewRegisterMemberCmd() *cobra.Command {
-	var hostKubeconfig, memberKubeconfig string
+	commandArgs := newRegisterMemberArgs()
 	cmd := &cobra.Command{
 		Use:   "register-member",
 		Short: "Executes add-cluster.sh script",
 		Long:  `Downloads the 'add-cluster.sh' script from the 'toolchain-cicd' repo and calls it twice: once to register the Host cluster in the Member cluster and once to register the Member cluster in the host cluster.`,
 		RunE: func(cmd *cobra.Command, args []string) error {
 			term := ioutils.NewTerminal(cmd.InOrStdin, cmd.OutOrStdout)
-			ctx := clicontext.NewCommandContext(term, client.DefaultNewClient)
+			ctx := newExtendedCommandContext(term, client.DefaultNewClientFromRestConfig)
 			newCommand := func(name string, args ...string) *exec.Cmd {
 				return exec.Command(name, args...)
 			}
-			return registerMemberCluster(ctx, newCommand, hostKubeconfig, memberKubeconfig)
+			return registerMemberCluster(ctx, newCommand, 5*time.Minute, commandArgs)
 		},
 	}
-	defaultKubeconfigPath := ""
-	if home := homedir.HomeDir(); home != "" {
-		defaultKubeconfigPath = filepath.Join(home, ".kube", "config")
-	}
-	cmd.Flags().StringVar(&hostKubeconfig, "host-kubeconfig", defaultKubeconfigPath, "Path to the kubeconfig file of the host cluster (default: "+defaultKubeconfigPath+")")
-	flags.MustMarkRequired(cmd, "host-kubeconfig")
-	cmd.Flags().StringVar(&memberKubeconfig, "member-kubeconfig", defaultKubeconfigPath, "Path to the kubeconfig file of the member cluster (default: "+defaultKubeconfigPath+")")
-	flags.MustMarkRequired(cmd, "member-kubeconfig")
+	cmd.Flags().StringVar(&commandArgs.hostKubeconfig, "host-kubeconfig", commandArgs.hostKubeconfig, fmt.Sprintf("Path to the kubeconfig file of the host cluster (default: %s)", commandArgs.hostKubeconfig))
+	cmd.Flags().StringVar(&commandArgs.memberKubeconfig, "member-kubeconfig", commandArgs.memberKubeconfig, fmt.Sprintf("Path to the kubeconfig file of the member cluster (default: %s)", commandArgs.memberKubeconfig))
+	cmd.Flags().BoolVar(&commandArgs.useLetsEncrypt, "lets-encrypt", commandArgs.useLetsEncrypt, fmt.Sprintf("Whether to use Let's Encrypt certificates or rely on the cluster certs (default: %t)", commandArgs.useLetsEncrypt))
+	cmd.Flags().Var(optionalUint{commandArgs.memberOrdinal}, "member-ordinal", "The explicit ordinal of the member used when there are multiple members in a single cluster. This can be used to re-register a member in case some changes or cleanup is needed. If unspecified, a new member with the next available ordinal is created, otherwise this is meant to represent the numerical suffix of the name with 0 meaning no suffix, 1 meaning suffix \"1\", etc. (default: unspecified)")
```

Comment:
I took me a while to understand your intentions. For the longest time I understood `register-member` to mean "register new member". What you wanted all that time was for the invocation to be idempotent and not try to deduce how the "new member" should be called. I will rework the logic (which should simplify a couple of things somewhat).

Reply:
So now the command should be idempotent and use the `--name-suffix` parameter to precisely control the suffix of the member name. I does not try to deduce some kind of new name to use when there is a collision and merely warns the user that it will overwrite an existing member.